legacyCode/app.js

```diff
@@ -1,54 +1,67 @@
 (function () {
 
-  var inputForm = document.getElementById("inputForm");
-  var buttonSearch = document.getElementById("buttonSearch");
-  var searchTerm = document.getElementById("searchTerm");
-  var loading = document.getElementById("loading");
-  var title = document.getElementById("title");
-  var inputDiv = document.getElementById("inputDiv");
-  var RepoList = document.getElementById("list");
+  var inputForm;
+  var RepoList;
+  var template;
+  var buttonSearch;
+  var searchTerm;
+  var loading;
+  var message;
 
   document.addEventListener("DOMContentLoaded", init, event);
 
   function init() {
+    message = document.getElementById("message");
+    inputForm = document.getElementById("inputForm");
+    buttonSearch = document.getElementById("buttonSearch");
+    searchTerm = document.getElementById("searchTerm");
+    loading = document.getElementById("loading");
+    RepoList = document.getElementById("list");
+    var source = document.getElementById("block-template").innerHTML;
     searchTerm.addEventListener("click", clearPage);
     inputForm.addEventListener("submit", enterSearchTerm);
     buttonSearch.addEventListener("click", clearList);
+    template = Handlebars.compile(source);
+
+    $(list).on("click", "a", function () {
+      var data = this.dataset;
+      GAE.modal.showRepoDetails(data.owner, data.name);
+    });
   }
 
   function enterSearchTerm(event) {
     event.preventDefault();
-    loading.innerText = "Loading...";
+    loading.setAttribute("class", "loader loading-position");
     GAE.services.requestRepos(searchTerm)
       .then(
         function success(repos) {
           showRepoList(repos);
-          loading.innerText = "";
+          loading.removeAttribute("class", "loader");
+          inputForm.removeAttribute("class", "start-input");
         },
         function error() {
-          Promise.reject("HTTP request is failed");
-          loading.innerText = "Error";
+          loading.removeAttribute("class", "loader");
+          message.innerText = "No repositories is found";
         }
       );
-    title.removeAttribute("class");
-    inputForm.removeAttribute("class");
   }
 
-  function showRepoList(items) {
-    items.forEach(function (repo) {
-      var li = list.appendChild(document.createElement('li'));
-      a = document.createElement('a');
-      a.href = repo.url;
-      a.innerHTML = repo.name;
-      li.appendChild(a);
+  function showRepoList(repos) {
+    var listContent = "";
+    repos.forEach(function (repo) {
+      var date = new Date(repo.updatedAt);
+      var context = {
+        Repo: repo,
+        repoDate: date.toLocaleDateString() + " " + date.getHours() + ":" + date.getMinutes()
```

Comment:
Need to format dates for UI using handlebars helpers. But can leave as is for now.